### ISSUE TITLE
lief >=0.13.0 won't install on centos:7 docker image running on Apple Mac M1

### ISSUE THREAD

**Issue by maayan-appdome** [lief-project/LIEF#986]
**Describe the bug**
when trying to install lief-0.14.0.dev0 on centos:7 docker image, I tried all .whl files in the latest repo, it says, for all of them,
"[.whl file] is not a supported wheel on this platform."

**To Reproduce**

1. I use Apple Mac with M1 chip
2. download and install docker from here https://docs.docker.com/desktop/install/mac-install/
3. run Docker through Applications
4. docker pull centos:7 
5. docker run -i -t centos:7 /bin/bash
6. yum update -y
7. yum install -y python3
8. download all 30 .whl files here https://lief.s3-website.fr-par.scw.cloud/latest/lief/, for all platforms
9. for each, run pip3 install [.whl file]

**Expected behavior**
one of the .whl file should have been installed successfuly

**Environment (please complete the following information):**
 - System and Version : Darwin 22.5.0 Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:20 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T6000 arm64

**Additional context**

**NOTICE**

If the issue does not contain enough information to be reproduced,
it will be flagged as [incomplete](https://github.com/lief-project/LIEF/issues?q=is%3Aissue+is%3Aclosed+label%3AIncomplete)
and closed.

**/NOTICE**

**Comment by romainthomas** [comment]
The default version of Python 3 in CentOS 7 is **3.6**. This version reached its end-of-life in the end of 2021 so yes, there is no wheel for this version.

**Comment by maayan-appdome** [comment]
hey, thanks for the quick answer
We are running Python **3.9.10** on CentOS7
```
[root@ee991c61a93f /]# python3
Python 3.9.10 (main, Aug 24 2023, 19:01:09) 

[root@ee991c61a93f /]# pip3 install lief==0.13.1
ERROR: Could not find a version that satisfies the requirement lief==0.13.1 (from versions: 0.8.0.post7, 0.8.1.post1, 0.8.2.post1, 0.8.3.post3, 0.9.0, 0.10.0, 0.10.1, 0.11.0, 0.11.1, 0.11.2, 0.11.3, 0.11.4, 0.11.5, 0.12.0, 0.12.1, 0.12.2, 0.12.3)
ERROR: No matching distribution found for lief==0.13.1
```

**Comment by romainthomas** [comment]
CentOS 7 is shipped with ld version `2.17` which was released in 2012. LIEF requires at least ld version 28 (released in 2018) and there is not plan to support `ld 2.17`
